app/decorators/coercion_decorator.rb

```diff
@@ -0,0 +1,38 @@
+class CoercionDecorator
+  extend Dry::Initializer
+
+  param :input, type: Types::Strict::Array.of(Types::Strict::Hash)
+
+  # @return [Array<Hash>]
+  def call
+    input.each { |row| row.each { |key, value| row[key] = format_value(key, value) } }
+    input
+  end
+
+private
+
+  # @param key [Symbol]
+  # @param value [Mixed]
+  # @return [Mixed]
+  def format_value(key, value)
+    if value.is_a?(Time)
+      format_datetime(value)
+    elsif key.to_s.include?('percentage')
+      format_percentage(value)
+    else
+      value
+    end
+  end
+
+  # @param element [Numeric]
+  # @return [String]
+  def format_percentage(element)
+    "#{(element * 100).round(2)}%"
+  end
+
+  # @param element [Time]
+  # @return [String]
+  def format_datetime(element)
```

Comment:
Taken as a whole I think it would be more appropriate to name the input `value` and the method `format_time`, would a `DateTime` get skipped?

Reply:
[0c368b6](https://github.com/DFE-Digital/early-years-foundation-recovery/pull/648/commits/0c368b6eb789a1cf6247a6bd7d09d4d90a2cfd9b)